starter-code/src/functions-and-arrays.js

```diff
@@ -11,10 +26,69 @@ var words = [
   'crackpot'
 ];
 
+// function findLongestWord(words) {
+//   return words.length;
+// }
+// findLongestWord(words);
+
+
+// if (findLongestWord(words)=== 0) {
+//   console.log("undefined");
+// }
+// else if (findLongestWord === 1 ){
+  
+//   console.log("foo");
+// }
+// findLongestWord(words); 
+
+
+
+function findLongestWord(wordsArr) {
+  var length = 0;
+  var longestWord;
+  for(var i=0; i< wordsArr.length; i++){
+    if(wordsArr[i].length > length){
+      longestWord = wordsArr[i];
+      length = wordsArr[i].length;
+    }
+  }
+  return longestWord;
+}
+
+ console.log("longest word is: ", findLongestWord(words));
+
+
+
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+
+function sumArray(){
+
+  var reducer = (accumulator, currentValue) => accumulator + currentValue;
```

Comment:
this function did not work the way you wanted it to. in order to use reduce you would do something like this
`
sumArray(array) {
return array.reduce(function (currentValue, accumulator) {
return currentValue + accumulator;
  }, 0)
}
`